coderd/experiments.go

```diff
@@ -4,16 +4,25 @@ import (
 	"net/http"
 
 	"github.com/coder/coder/v2/coderd/httpapi"
+	"github.com/coder/coder/v2/codersdk"
 )
 
 // @Summary Get experiments
 // @ID get-experiments
 // @Security CoderSessionToken
 // @Produce json
 // @Tags General
+// @Param include_all query bool false "All available experiments"
 // @Success 200 {array} codersdk.Experiment
 // @Router /experiments [get]
 func (api *API) handleExperimentsGet(rw http.ResponseWriter, r *http.Request) {
 	ctx := r.Context()
-	httpapi.Write(ctx, rw, http.StatusOK, api.Experiments)
+	all := r.URL.Query().Has("include_all")
+
+	if !all {
+		httpapi.Write(ctx, rw, http.StatusOK, api.Experiments)
+		return
+	}
+
+	httpapi.Write(ctx, rw, http.StatusOK, codersdk.ExperimentsAll)
```

Comment:
It is more code to just always return both pieces of information, however if we are going to split it, it might be better just to make it different routes, rather than a query param.

Like `/experiments/available` and `/experiements`. The query param feels a bit strange to me imo. And `include_all` is omitting the experiments set that are not included, so it's switching behavior, rather than extending.

Reply:
Since we aren't positive we want to keep this list as per the discussion in the ticket, I wasn't sure it made sense to spin up a whole new route, although I can if you feel strongly about it. This endpoint is only called for this one feature.

> And `include_all` is omitting the experiments set that are not included, so it's switching behavior, rather than extending.

`include_all` would include the entire experiment set, enabled or not. Am I misunderstanding your comment?

Reply:
Ah, you mean it doesn't include unsafe experiments. What if we renamed it to `include_all_safe`?

Reply:
If we do not want to merge the responses, I think we should just make another endpoint:
```golang
func (api *API) handleExperimentsSafe(rw http.ResponseWriter, r *http.Request) {
	ctx := r.Context()
	
	httpapi.Write(ctx, rw, http.StatusOK, codersdk.ExperimentsAll)
}

r.Route("/experiments", func(r chi.Router) {
	r.Use(apiKeyMiddleware)
	r.Get("/available", api.handleExperimentsSafe)
	r.Get("/", api.handleExperimentsGet)
})
```

Reply:
Updated!